core/state_transition.go

```diff
@@ -190,12 +190,13 @@ func IntrinsicGas(data []byte, contractCreation bool, feeCurrency *common.Addres
 }
 
 // NewStateTransition initialises and returns a new state transition object.
-func NewStateTransition(evm *vm.EVM, msg Message, gp *GasPool) *StateTransition {
-	gasPriceMinimum, _ := gpm.GetGasPriceMinimum(msg.FeeCurrency(), evm.Header, evm.StateDB)
+func NewStateTransition(evm *vm.EVM, msg Message, gp *GasPool, caller vm.EVMCaller) *StateTransition {
```

Comment:
This is confusing, why take an evm and a caller?

Reply:
I know, any ideas?

i can't really use the evm to construct the evmCaller; nor i can get a new evm from the caller. Since they are different, they have different vm.Context object.

Reply:
Whats the difference in the vm.Contexts? Can we share one instead?

Reply:
So I looked into this a bit more and I think all that differs between the evms is the sender and the gas price. So we could pass a factory here and provide a method on the factory to build an evm taking the caller, gas price, header and state. Then we would just need that one factory instance. This would change the meaning of the EVMCaller since it would no longer be an object reserved solely for calls by the system, but I think is a good change since then I think it supports all our use cases and we don't end up with 2 different ways of calling the evm.

Reply:
i would wait on that... actually i've chatter with @oneeman before making this change since a faced two paths:

 * Add a argument and minimize changes that would conflict upstream
 * Do something more fancy like you suggest, and complicate upstream merges

And our "merge master" discouraged that approach. 

And since that would be a longer conversations i would leave this here as it is now, and we can get back at it later.